kubernetes-model-generator/kubernetes-model-core/src/main/java/io/fabric8/kubernetes/api/model/HasMetadata.java

```diff
@@ -232,4 +235,109 @@ default boolean isFinalizerValid(String finalizer) {
   default boolean removeFinalizer(String finalizer) {
     return getMetadata().getFinalizers().remove(finalizer);
   }
+
+  default boolean hasOwnerReferenceFor(HasMetadata owner) {
+    return getOwnerReferenceFor(owner).isPresent();
+  }
+
+  default boolean hasOwnerReferenceFor(String ownerUid) {
+    return getOwnerReferenceFor(ownerUid).isPresent();
+  }
+
+  default Optional<OwnerReference> getOwnerReferenceFor(HasMetadata owner) {
+    if (owner == null) {
+      return Optional.empty();
+    }
+
+    final String ownerUID = owner.getMetadata().getUid();
+    return getOwnerReferenceFor(ownerUID);
+  }
+
+  default Optional<OwnerReference> getOwnerReferenceFor(String ownerUid) {
+    if (ownerUid == null || ownerUid.isEmpty()) {
+      return Optional.empty();
+    }
+
+    return getMetadata().getOwnerReferences().stream()
```

Comment:
The owner references could be null.  Granted it would require someone calling setOwnerReferences(null).  Maybe the code generator could guard against that.